core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/stack/PaywallCredentialValidator.java

```diff
@@ -0,0 +1,86 @@
+package com.sequenceiq.cloudbreak.controller.validation.stack;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.model.AmbariRepoDetailsJson;
+import com.sequenceiq.cloudbreak.api.model.AmbariStackDetailsJson;
+import com.sequenceiq.cloudbreak.api.model.stack.cluster.ClusterRequest;
+import com.sequenceiq.cloudbreak.cloud.model.component.AmbariInfo;
+import com.sequenceiq.cloudbreak.cloud.model.component.DefaultHDFEntries;
+import com.sequenceiq.cloudbreak.cloud.model.component.DefaultHDPEntries;
+import com.sequenceiq.cloudbreak.cloud.model.component.StackInfo;
+import com.sequenceiq.cloudbreak.controller.exception.BadRequestException;
+import com.sequenceiq.cloudbreak.service.DefaultAmbariRepoService;
+import com.sequenceiq.cloudbreak.service.credential.PaywallCredentialService;
+
+@Component
+public class PaywallCredentialValidator {
+
+    @Inject
+    private PaywallCredentialService paywallCredentialService;
+
+    @Inject
+    private DefaultHDPEntries defaultHDPEntries;
+
+    @Inject
+    private DefaultHDFEntries defaultHDFEntries;
+
+    @Inject
+    private DefaultAmbariRepoService defaultAmbariRepoService;
+
+    public void validateCredential(ClusterRequest clusterRequest, String stackType, String stackVersion) {
+        StackInfo stackInfo = getStackInfo(stackType, stackVersion);
+        boolean noPaywallCredentialAvailable = noPaywallCredentialAvailable();
+        if (clusterRequest.getAmbariStackDetails() != null && clusterRequest.getAmbariRepoDetailsJson() != null) {
+            if ((isStackPaywallProtected(clusterRequest, stackInfo) || isAmbariPaywallProtected(clusterRequest)) && noPaywallCredentialAvailable) {
+                throwException();
+            }
+        } else if (stackInfo != null && stackInfo.isPaywallProtected() && noPaywallCredentialAvailable) {
```

Comment:
isStackPaywallProtected() could check this setting, too, as it already checks almost the same logic